jekyll/_cci2/api-developers-guide.md

```diff
@@ -0,0 +1,563 @@
+---
+layout: classic-docs
+title: "CircleCI API Developer's Guide"
+short-title: "Developer's Guide"
+description: "API cookbook for internal and external CircleCI developers"
+categories: [getting-started]
+order: 1
+---
+
+This *API Developer's Guide* was written to assist developers in quickly and easily making API calls to CircleCI services to return detailed information about users, pipelines, projects and workflows.
+
+* TOC
+{:toc}
+
+# API Overview
+
+The CircleCI platform provides a powerful API that enables users to retrieve detailed information about users, jobs, workflows and pipelines.
+
+## API Classes
+
+The current classes of the API v2 endpoints are:
+
+* Authentication
+* Pipeline
+* Workflows
+* User (Preview)
+* Project (Preview)
+* Job (Preview)
+
```

Comment:
Should we add the new Insights API in preview?
https://circleci.com/api/v2/insights
CC @KunalJain

Reply:
@KyleTryon We will add Insights API endpoints in Q1.